### PR TITLE
Add encoding and decoding support for SC address

### LINKED ISSUE #76: Add support for SC (SIA) addresses in encoding library
**UPDATE: This was previous closed since SC (SIA) didn't have a SLIP-0044 identifier, but [now it does](https://github.com/satoshilabs/slips/blob/master/slip-0044.md).

Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

### PR DESCRIPTION
**Pull request by janus**
This is  PR adds support for SC (SIA) addresses in encoding library

## Issue number

closes #76
## Description

Add scAddressEncoder and scAddressDecoder function in index.ts file. In the test file, I added a  test case.
## Reference to the specification
https://sia.tech/docs/#wallet-address-get
https://gitlab.com/NebulousLabs/Sia/-/blob/master/types/encoding.go#L910

## Reference to the test address.

https://sc.tokenview.com/
Got my test example from the above link

## List of features added/changed

- [x] No import just a basic function call

NOTE: If you are adding new coin address support, please make sure to add a reference link to README so that reviewer can verify.

## How much has the filesize increased?
345380

## How Has This Been Tested?

I ran `npm test`.  I used an already provided test setup, just added my test case

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.
- [x] I have specified correct coinTypes specified at [Slip 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- [x] I have provided the reference link to the specification I implemented.
- [x] I have provided enough explanation about how I provided the test address

### CODE REVIEW

**Comment by janus** [comment]
@makoto Please review

File: src/index.ts [hunk 1]
```diff
@@ -791,6 +791,24 @@ function ardrAddressEncoder(data: Buffer): string {
...
+
+function scAddressDecoder(data: string): Buffer {
+  if(data.length !== 76) {
+    throw Error('Unrecognised address format');
+  }
+
+  return Buffer.from(data, 'hex');
+}
+
+
```
Comment (makoto) [inline src/index.ts:811]: Please remove extra new line

Comment (janus) [reply src/index.ts]: Thanks. Resolved. Comment.

**Comment by makoto** [comment]
Note to self, the file size is 346623 -> 377973, about 30k growth

**Comment by makoto** [comment]
Spoke to Sia team and they mentioned that it may need checksum. Waiting for the team to clarify.

File: src/index.ts [hunk 2]
```diff
@@ -791,6 +791,21 @@ function ardrAddressEncoder(data: Buffer): string {
 
 }
 
+function scAddressEncoder(data:Buffer): string {
+  const hexString = data.toString('hex');
+  if(hexString.length !== 76) {
```
Comment (peterjanbrone) [inline src/index.ts:796]: A siacoin address is a hash + checksum, in hex representation. So aside from the length check, we should also add a checksum verification here. If you take a look at [this function](https://gitlab.com/NebulousLabs/Sia/-/blob/master/types/encoding.go#L910) it should become clear on what's going on. Both that function and the `String` function are of interest here, that's basically going from and to a siacoin address and string. Note that we use a blake2b hashing function.

Comment (janus) [reply src/index.ts:962]: Thanks.

**Comment by janus** [comment]
@makoto  Please review and merge.

**Comment by makoto** [comment]
Hi, I will be merging [another PR](https://github.com/ensdomains/address-encoder/pull/289) as his PR came before you added the checksum support and also I prefer his coding standard (better use of constants and reuse of `blake2bHex`).

**Comment by janus** [comment]
> Hi, I will be merging [another PR](https://github.com/ensdomains/address-encoder/pull/289) as his PR came before you added the checksum support and also I prefer his coding standard (better use of constants and reuse of `blake2bHex`).

No problem....

**Comment by janus** [comment]
His came not before the update from SIA source. I was already on this and you told me to wait for the update,  and while holidaying I pushed my update. 

BTW, thanks for everything.

**Pull request closed without merging**